plugins/mumble_plugin_utils.h

```diff
@@ -28,14 +28,36 @@ union SingleSplit4Bytes {
 	constexpr SingleSplit4Bytes(const uint32_t value) : single(value) {}
 };
 
-static inline std::string utf16ToUtf8(const std::wstring &wstr) {
-	std::wstring_convert< std::codecvt_utf8_utf16< wchar_t > > conv;
-	return conv.to_bytes(wstr);
+/// Converts from UTF-8 to UTF-16.
+static inline std::wstring utf8ToUtf16(const std::string &str) {
+	try {
+		std::wstring_convert< std::codecvt_utf8_utf16< wchar_t > > conv;
+		return conv.from_bytes(str);
+	} catch (std::range_error &) {
+		return {};
+	}
 }
 
-static inline std::wstring utf8ToUtf16(const std::string &str) {
-	std::wstring_convert< std::codecvt_utf8_utf16< wchar_t > > conv;
-	return conv.from_bytes(str);
+/// Converts from UTF-16 to UTF-8.
+/// Meant to be used with Windows processes, where "wchar_t" is 2 bytes.
+static inline std::string utf16ToUtf8(const std::u16string &str) {
+	try {
+		std::wstring_convert< std::codecvt_utf8_utf16< char16_t >, char16_t > conv;
+		return conv.to_bytes(str);
+	} catch (std::range_error &) {
+		return {};
+	}
+}
+
+/// Converts from UTF-16 to UTF-8.
+/// Meant to be used with Linux processes, where "wchar_t" is 4 bytes.
+static inline std::string utf16ToUtf8(const std::u32string &str) {
+	try {
+		std::wstring_convert< std::codecvt_utf8_utf16< char32_t >, char32_t > conv;
+		return conv.to_bytes(str);
+	} catch (std::range_error &) {
+		return {};
+	}
```

Comment:
```suggestion
static inline std::string utf16ToUtf8_win(const std::u16string &str) {
	try {
		std::wstring_convert< std::codecvt_utf8_utf16< char16_t >, char16_t > conv;
		return conv.to_bytes(str);
	} catch (std::range_error &) {
		return {};
	}
}

/// Converts from UTF-16 to UTF-8.
/// Meant to be used with Linux processes, where "wchar_t" is 4 bytes.
static inline std::string utf16ToUtf8_unix(const std::u32string &str) {
	try {
		std::wstring_convert< std::codecvt_utf8_utf16< char32_t >, char32_t > conv;
		return conv.to_bytes(str);
	} catch (std::range_error &) {
		return {};
	}
```

Otherwise it's way too easy to mess up an d use the wrong function

Reply:
It's actually not, the compiler chooses the function depending on the argument.

The main reason why I removed the function that accepted `std::wstring` is to make sure that the character type size is enforced.

Also, keep in mind that a game might for example use `char16_t` for both Linux and Windows binaries.

Reply:
Ah hold on. I didn't check the parameter types properly. On second thought I agree

Reply:
No problem.

I'm going to improve the comment a bit to make it clear that the size is not guaranteed even on the same platform.